api/metrics/legacy/http.go

```diff
@@ -79,6 +84,13 @@ func WithGlobalMiddleware(m ...func(http.Handler) http.Handler) HandlerOption {
 	}
 }
 
+// WithLabelParser adds a custom label parser to the handler.
```

Comment:
Maybe a line on what the label parser aims to do here would be great?

Reply:
done